doc/supervised.rst

```diff
@@ -41,32 +41,43 @@ two numbers.
 
 Fit, transform, and so on
 -------------------------
-The goal of supervised metric-learning algorithms is to transform
+Generally, the goal of supervised metric-learning algorithms is to transform
 points in a new space, in which the distance between two points from the
 same class will be small, and the distance between two points from different
-classes will be large. To do so, we fit the metric learner (example:
-`NCA`).
+classes will be large.
+
+But there are also some algorithms that learn a similarity, not a distance,
+thus the points cannot be transformed into a new space. In this case, the
+utility comes at using the similarity bewtween points directly.
+
+Mahalanobis learners can transform points into a new space, and to do so,
+we fit the metric learner (example:`NCA`).
```

Comment:
We need to reformulate this: the initial sentence is too restrictive now, and keeping it induces some confusion.

The goal of supervised metric learning algorithms is to learn a (distance or similarity) metric such that two points from the same class will be similar (e.g., have small distance) and points from different classes will be dissimilar (e.g., have large distance).

To do so, we first need to fit the supervised metric learner on a labeled dataset, as in the example below with `NCA`.

Reply:
Changed.